src/tiledb/cloud/files/ingestion.py

```diff
@@ -281,8 +284,249 @@ def ingest_files(
+    # Graph Setup
+    taskgraph_name = taskgraph_name or "files-ingest-and-index"
+    graph = dag.DAG(
+        name=taskgraph_name,
+        namespace=namespace,
+        mode=Mode.BATCH,
+    )
+
+    # Step 1: File ingestion graph
+    ingestion = graph.submit(
+        ingest_files,
+        dataset_uri,
+        search_uri=search_uri,
+        pattern=pattern,
+        ignore=ignore,
+        max_files=max_files,
+        batch_size=batch_size,
+        acn=acn,
+        config=config,
+        namespace=namespace,
+        group_uri=group_uri,
+        taskgraph_name=f"initiated from: ({taskgraph_name})",
+        ingest_resources=ingest_resources,
+        verbose=verbose,
+    )
+
+    # Step 2: File indexing graph
+    indexing = graph.submit(
+        index_files,
+        # Search in the ingested files' uri
+        search_uri=dataset_uri,
+        index_uri=index_uri,
+        acn=acn,
+        config=config,
+        environment_variables=environment_variables,
+        namespace=namespace,
+        verbose=verbose,
+        trace_id=trace_id,
+        index_type=index_type,
+        index_creation_kwargs=index_creation_kwargs,
+        index_dag_resources=index_dag_resources,
+        include=pattern,
+        exclude=ignore,
+        suffixes=suffixes,
+        max_files=max_files,
+        text_splitter=text_splitter,
+        text_splitter_kwargs=text_splitter_kwargs,
+        embedding_class=embedding_class,
+        embedding_kwargs=embedding_kwargs,
+        openai_key=openai_key,
+        index_timestamp=index_timestamp,
+        workers=workers,
+        worker_resources=worker_resources,
+        worker_image=worker_image,
+        extra_worker_modules=extra_worker_modules,
+        driver_resources=driver_resources,
+        driver_image=driver_image,
+        extra_driver_modules=extra_driver_modules,
+        max_tasks_per_stage=max_tasks_per_stage,
+        embeddings_generation_mode=embeddings_generation_mode,
+        embeddings_generation_driver_mode=embeddings_generation_driver_mode,
+        vector_indexing_mode=vector_indexing_mode,
+        index_update_kwargs=index_update_kwargs,
+        threads=threads,
+        ingest_resources=index_resources,
+        consolidate_partition_resources=consolidate_partition_resources,
+        copy_centroids_resources=copy_centroids_resources,
+        random_sample_resources=random_sample_resources,
+        kmeans_resources=kmeans_resources,
+        compute_new_centroids_resources=compute_new_centroids_resources,
+        assign_points_and_partial_new_centroids_resources=assign_points_and_partial_new_centroids_resources,
+        write_centroids_resources=write_centroids_resources,
+        partial_index_resources=partial_index_resources,
+    )
+    indexing.depends_on(ingestion)
+
+    # Start the ingestion and indexing process
+    run_dag(graph, wait=wait, debug=verbose)
+    return str(graph.server_graph_uuid)
+
+
+ingest_and_index = as_batch(ingest_and_index)
```

Comment:
Do we need to use different names to expose both the raw function and the `as_batch` version?

Reply:
I am doing this to follow the VCF's implementation according to ingestion.
But at some point we need to refactor the ingestion methods.
I am thinking something along the lines of Factory for ingestors and Strategy for ingestor selection